### ISSUE TITLE
2-step authentication

### ISSUE THREAD

**Issue by mbsmt** [Terrorhawk/Capri#18]
Is there not 2-step authentication options in this skin?

**Comment by beliys** [comment]
Thank you for request. Please update and check.

**Comment by mbsmt** [comment]
Thank you Beliys. I've updated the skin and everything is find now, specially 2-step authentication.

**Comment by mbsmt** [comment]
It seems your skin is so heavy to load. I used Capri skin before too (not your github version), and it was so fast. But as I installed this version of Capri, a simple login and showing first page after that may take about 2-3 minutes (even with missing page elements).

**Comment by beliys** [comment]
On my server skin load very fast. I need debug for check problem.

**Comment by mbsmt** [comment]
What can I do for it ?

**Comment by beliys** [comment]
Do you can give me test account for directadmin and console? If you can than sent him to beliy@yandex.ru

**Comment by mbsmt** [comment]
Unfortunately I can't. Is there anything that I can do myself?

**Comment by beliys** [comment]
Do you have error in browser console? Have error in php debug mode?

**Comment by beliys** [comment]
What elements be missing?

**Comment by mbsmt** [comment]
For example after first login, I waited for seeing 2-step authentication section about 3 minutes. And finally it did not load. I had to refresh the page 3 or 4 times to watch it.

**Comment by beliys** [comment]
Do you have error in browser console?

**Comment by mbsmt** [comment]
Nothing ...

**Comment by mbsmt** [comment]
Maybe it was a connection problem at that time. I will check errors once again if I occur with this problem.
Thank you for your attention.

**Comment by beliys** [comment]
I fix potential problem with two-step auth (can load slow after login). Please update.

**Comment by mbsmt** [comment]
Before update, I took a look at console errors in browser and these were the errors:

Failed to load resource: net::ERR_TIMED_OUT
for IMG_MENU_HELP and IMG_LV_BT_ADMIN

What is the last version right now?

**Comment by beliys** [comment]
i think problem be in domain boxes. Where use API function before full auth (before login security questions ). Directadmin not return anything  for non auth user and you wait php timeout for this function.  I add temporarily check for this block (|*if HAVE_TWOSTEP_AUTH!="yes"|) and be thinking how better global fix it.

**Comment by mbsmt** [comment]
Yes, I think so. Delay to show UI is between first login (successful) and asking 2-step authentication code.

By the way, I am not sure could I update to latest version or not. May you take a look at the attached image please 
![capture](https://cloud.githubusercontent.com/assets/8167221/10752707/f33070e4-7ca0-11e5-9396-e612e45e99cf.JPG)
?

**Comment by beliys** [comment]
last version is 1.33 but you have 1.32
for check version skin use next php function
```php
if($versions = @file_get_contents("https://raw.githubusercontent.com/Terrorhawk/Capri/master/version.txt")) {
    list($ava_version, $da_version) = explode('-', $versions);
  } else {
    $ava_version = 'n/a';
    $da_version = 'n/a';
  }
```
you have error in file_get_contents("https://raw.githubusercontent.com/Terrorhawk/Capri/master/version.txt"). Check work php and network on you server for understanding problem.

**Comment by Terrorhawk** [comment]
I found somthing about the slow loading.

It only happends to me in firefox. sometimes a page loads realy slow in firefox.
In Chrome i never seen this loading time.
Did not test IE so dont know if it happends there

**Comment by mbsmt** [comment]
@Terrorhawk I've checked it in Chrome only and saw slow process to show 2-step authentication form.
@beliys You were right. Update problem belongs to `file_get_contents("https://raw.githubusercontent.com/Terrorhawk/Capri/master/version.txt")`. Unfortunatelly https://raw.githubusercontent.com seems blocked connections from some countries (Like Iran, where I am). Is there not any other option or repository?

**Comment by beliys** [comment]
@Terrorhawk 
im debuging skin in firefox and dont have any problems. How pages slow load? How errors in browser console? Have errors if turn on debug php mode? Have errors in directadmin log? Have error if disable all firefox extentions?

@mbsmt 
You problem with slow 2 step form be fixed in last version... Need update...
How you write here with blocked connection?
You can download and update skin manually or setting up tor, vpn or any another anonymous service.

**Comment by mbsmt** [comment]
@beliys github.com is open for me, what I said is blocking me was https://raw.githubusercontent.com/.
I hope I can solve this problem in some way.

**Comment by mbsmt** [comment]
@beliys I installed version 1.33 and it seems slow processing to show 2-step authentication form is solved completely. Up to now, I did not see that delay again ...

Thank you.

**Comment by mbsmt** [comment]
@beliys as a suggestion, I think it can be so good if Capri can detect 2-step authentication status (Is it enable or disable) and focus on authentication code field automatically after page load.
